### PR TITLE
SRVLOGIC-382: SonataFlow Management Console productization setup in midstream

### PR DESCRIPTION
**Pull request by fantonangeli**
**Jira:** https://issues.redhat.com/browse/SRVLOGIC-382

**Description:**
Create a setup to productize SonataFlow Management Console in midstream kie-tools repo

**Notes:**
@ricardozanini This configuration creates a zip from the dist folder. The destination file is:
`packages/sonataflow-management-console-webapp/target/sonataflow-management-console-webapp-999-SNAPSHOT-image-build.zip`

### CODE REVIEW

File: packages/sonataflow-management-console-webapp/pom.xml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+              >-F sonataflow-management-console-webapp... -F !@kie-tools/dmn-marshaller-backend-compatibility-tester -F !@kie-tools/dmn-testing-models build:prod</arguments>
+              <environmentVariables>
+                <KIE_TOOLS_BUILD__runTests>false</KIE_TOOLS_BUILD__runTests>
+                <KIE_TOOLS_BUILD__runIntegrationTests>false</KIE_TOOLS_BUILD__runIntegrationTests>
+                <SONATAFLOW_MANAGEMENT_CONSOLE_WEBAPP__sonataflowEnvMode
+                >true</SONATAFLOW_MANAGEMENT_CONSOLE_WEBAPP__sonataflowEnvMode>
+              </environmentVariables>
+            </configuration>
+          </execution>
+        </executions>
```
Comment (domhanak) [inline packages/sonataflow-management-console-webapp/pom.xml:74]: This looks a bit strange @rgdoliveira

File: packages/sonataflow-management-console-webapp/pom.xml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      <plugin>
+        <groupId>com.github.eirslett</groupId>
+        <artifactId>frontend-maven-plugin</artifactId>
+        <inherited>false</inherited>
+        <configuration>
+          <workingDirectory>../../</workingDirectory>
+        </configuration>
+        <executions>
+          <execution>
+            <id>pnpm build:prod</id>
```
Comment (domhanak) [inline packages/sonataflow-management-console-webapp/pom.xml:57]: @fantonangeli before we build:prod, shouldn't we also do `pnpm bootstrap -F "sonataflow-management-console-webapp..."`

File: packages/sonataflow-management-console-webapp/pom.xml (hunk 1 above)
Comment (fantonangeli) [reply packages/sonataflow-management-console-webapp/pom.xml:74]: @domhanak I tried again now to fix this formatting, but I think it is the linter, called by the git hooks, that makes this formatting.
See lines 64, 68 and 69

Comment (domhanak) [reply packages/sonataflow-management-console-webapp/pom.xml:74]: Thanks for the pointer, if it building I am all good. Just wanted to double check.

File: packages/sonataflow-management-console-webapp/pom.xml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            <goals>
+              <goal>pnpm</goal>
+            </goals>
+            <configuration>
+              <skip>${skip.ui.build}</skip>
+              <pnpmInheritsProxyConfigFromMaven>false</pnpmInheritsProxyConfigFromMaven>
+              <arguments
+              >-F sonataflow-management-console-webapp... -F !@kie-tools/dmn-marshaller-backend-compatibility-tester -F !@kie-tools/dmn-testing-models build:prod</arguments>
```
[the remark this answers is not in the document]
Comment (fantonangeli) [reply packages/sonataflow-management-console-webapp/pom.xml:65]: The "@kie-tools/" prefix was needed in the previous version of pnpm and we do the same here:
https://github.com/kiegroup/kie-tools/blob/59c4084505bfe27b1d862449edb4f0f43f486442/packages/sonataflow-quarkus-devui/pom.xml#L99

See also: https://github.com/apache/incubator-kie-tools/blob/1e6b21e7e9f657c99d94050354f88eba19471375/README.md?plain=1#L69

Maybe it's more consistent to put the prefix in both poms?
I can do this in this PR

File: packages/sonataflow-management-console-webapp/pom.xml (hunk 2 above)
Comment (fantonangeli) [reply packages/sonataflow-management-console-webapp/pom.xml:57]: The bootstrap it's executed here: https://github.com/kiegroup/kie-tools/pull/39/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8

File: packages/sonataflow-management-console-webapp/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (domhanak) [reply packages/sonataflow-management-console-webapp/pom.xml:65]: Oh nice, didn't catch the prefix is now not needed, My comment originates from the fact we have the `@kie-tools` prefix still in many readmes in community. So no need to change here, I think this is clarified.

[the remark this answers is not in the document]
Comment (domhanak) [reply packages/sonataflow-management-console-webapp/pom.xml:65]: https://github.com/apache/incubator-kie-tools/blob/1e6b21e7e9f657c99d94050354f88eba19471375/packages/sonataflow-management-console-image/README.md example

**Review by domhanak — APPROVED** [review]

File: packages/sonataflow-management-console-webapp/pom.xml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+              <environmentVariables>
+                <KIE_TOOLS_BUILD__runTests>false</KIE_TOOLS_BUILD__runTests>
+                <KIE_TOOLS_BUILD__runIntegrationTests>false</KIE_TOOLS_BUILD__runIntegrationTests>
+                <SONATAFLOW_MANAGEMENT_CONSOLE_WEBAPP__sonataflowEnvMode
+                >true</SONATAFLOW_MANAGEMENT_CONSOLE_WEBAPP__sonataflowEnvMode>
```
Comment (ricardozanini) [inline packages/sonataflow-management-console-webapp/pom.xml:70]: This line break here won't cause any issues? Same in the line 65.

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**